dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -76,6 +77,13 @@
         emptyBodyDeserializer = new EmptyBodyDeserializer(errorDecoder);
     }
 
+    private static List<WeightedEncoding> decorateEncodings(List<WeightedEncoding> input) {
```

Comment:
lets do this wrapping outside of the conjureBodySerde?

Reply:
not sure that makes sense, because users can in theory supply their own encodings using the builder?

Reply:
There's an argument for using minimal logic here and providing memoizing encoding wrappers from WC, which is why I hadn't proposed this yet. Since we've started looking at memoizing these things, I think it's best to centralize memoization in one place than to mix it, and this hedges the worst case (many large services where only a small subset of endpoints are used)